src/kernel/src/dev/dipsw.rs

```diff
@@ -0,0 +1,39 @@
+use crate::{
+    errno::{Errno, EINVAL},
+    fs::{Device, IoCmd},
+    process::VThread,
+};
+use macros::Errno;
+use std::sync::Arc;
+use thiserror::Error;
+
+#[derive(Debug)]
+struct Dipsw {}
+
+impl Device for Dipsw {
+    fn ioctl(self: Arc<Self>, cmd: IoCmd, td: &VThread) -> Result<(), Box<dyn Errno>> {
+        if !td.cred().is_system() {
+            return Err(Box::new(IoctlErr::InsufficientCredentials));
```

Comment:
What I saw is there are some commands that can be execute without a system credential.

Reply:
Soooo.... Option\<&VThread\>?

Reply:
I mean there are some commands that can be execute if `sceSblACMgrIsSystemUcred` return 0. So return error here is not correct.

Reply:
Oh :eyes:

Reply:
You're right. There's so many nested ifs that I overlooked that. Fixed